utils/disk_space_monitor.cc

```diff
@@ -30,6 +30,28 @@ disk_space_monitor::disk_space_monitor(abort_source& as, std::filesystem::path d
     , _signal_barrier("disk_space_monitor::signal_barrier")
     , _data_dir(std::move(data_dir))
     , _cfg(std::move(cfg))
+    , _threshold_subscription(listen([this](const disk_space_monitor& dsm) -> future<> {
+        const float current_disk_utilization = dsm.disk_utilization();
+        if (current_disk_utilization < 0.0f) {
+            co_return;
+        }
+
+        dsmlog.debug("current disk utilization={}", current_disk_utilization);
+
+        for (auto& sub : _subscriptions) {
+            auto threshold_reached = above_threshold(current_disk_utilization > std::clamp(sub._threshold(), 0.0f, 1.0f));
+
+            const bool crossed_threshold = (sub._threshold_reached != threshold_reached);
+            const bool constant_update = !sub._trigger_options.only_crossing_threshold;
+
+            if (constant_update || crossed_threshold) {
+                if ((threshold_reached == above_threshold::yes && sub._trigger_options.when_above_threshold) ||
+                    (threshold_reached == above_threshold::no && sub._trigger_options.when_below_threshold)) {
+                    co_await sub(std::move(threshold_reached));
```

Comment:
what about yhe contant update case?
did I miss anything?
is there a test case for it?